File: Behavioral/Observer/Java/NewsAgency.java
// NewsAgency.java

import java.util.ArrayList;
import java.util.List;

public class NewsAgency implements Subject {
  private List<Observer> observers = new ArrayList<>();

  @Override
  public void addObserver(Observer observer) {
    observers.add(observer);
  }

  @Override
  public void removeObserver(Observer observer) {
    observers.remove(observer);
  }

  @Override
  public void notifyObservers(String news) {
    for (Observer observer : observers) {
      observer.update(news);
    }
  }
}

File: Behavioral/Observer/Python/news_agency.py
# news_agency.py

from subject import Subject

class NewsAgency(Subject):
  def __init__(self):
    self._observers = []

  def add_observer(self, observer):
    self._observers.append(observer)

  def remove_observer(self, observer):
    self._observers.remove(observer)

  def notify_observers(self, news):
    for observer in self._observers:
      observer.update(news)
